shell/store/index.js

```diff
@@ -124,6 +124,9 @@ const getActiveNamespaces = (state, getters) => {
       }
     }
   }
+  // Create map that can be used to efficiently check if a
+  // resource should be displayed
+  state.activeNamespaceCache = out;
```

Comment:
For safety this should tie in to `updateActiveNamespaceCache`.

Updating the `activeNamespaceCacheKey` value should probably move into getActiveNamespaces underneath `state.activeNamespaceCache = out;`. Then calls to `updateActiveNamespaceCache` can be removed and calls to it go straight to `getActiveNamespaces`

Reply:
> For safety this should tie in to updateActiveNamespaceCache.

No, this won't work. We can't call `updateNamespaceCache` from within `getActiveNamespaces` because `updateNamespaceCache` calls `activeNamespaces`. It results in an infinite loop that I have seen with my own eyes.

Reply:
That's not quite what i suggested. 

1. The following can be added to `getActiveNamespaces` below `state.activeNamespaceCache = out;`
```
  // This is going to run a lot, so keep it optimised
  let cacheKey = '';

  for (const key in state.activeNamespaceCache) {
    cacheKey += key + state.activeNamespaceCache[key];
  }
  state.activeNamespaceCacheKey = cacheKey;
```
2. updateActiveNamespaceCache can be removed
3. places that called updateActiveNamespaceCache can call getActiveNamespaces

Reply:
I've made that change, along with refactoring computed properties. It seems to work

Reply:
i missed a bug related to fleet, fixed in a commit i've just pushed. that re-works this a smidge to take into account the early exits from getActiveNamespaces